Feature `unstable-provider-api` unconditionally requires `std`

# Steps to reproduce

```console
$ cargo new snafu-test
    Creating binary (application) `snafu-test` package
note: see more `Cargo.toml` keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

$ cd snafu-test

$ cargo add snafu@0.8.8 --no-default-features -F unstable-provider-api -F rust_1_81
    Updating crates.io index
      Adding snafu v0.8.8 to dependencies
             Features:
             + rust_1_61
             + rust_1_65
             + rust_1_81
             + unstable-provider-api
             - alloc
             - backtrace
             - backtraces-impl-backtrace-crate
             - futures
             - futures-core-crate
             - futures-crate
             - guide
             - internal-dev-dependencies
             - pin-project
             - std
             - unstable-core-error
             - unstable-try-trait

$ cargo +nightly build -q
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `std`
   --> /home/nksm/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/snafu-0.8.8/src/report.rs:248:17
    |
248 |                 std::writeln!(f, "\nBacktrace:\n{}", bt)?;
    |                 ^^^ use of unresolved module or unlinked crate `std`
    |
note: found an item that was configured out
   --> /home/nksm/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/snafu-0.8.8/src/lib.rs:276:14
    |
275 | #[cfg(feature = "std")]
    |       --------------- the item is gated behind the `std` feature
276 | extern crate std;
    |              ^^^

For more information about this error, try `rustc --explain E0433`.
error: could not compile `snafu` (lib) due to 1 previous error


```

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -122,6 +122,22 @@ rust_1_81_no_std_test_task:
     - cargo build --target thumbv6m-none-eabi --no-default-features --features=rust_1_81,alloc,futures
   before_cache_script: rm -rf $CARGO_HOME/registry/index
 
+rust_nightly_no_std_test_task:
+  name: "Rust Nightly (no_std)"
+  container:
+    image: rustlang/rust:nightly
+    cpu: 1
+    memory: 2Gi
+  cargo_cache:
+    folder: $CARGO_HOME/registry
+    fingerprint_script: cat Cargo.toml
+  setup_script:
+    - rustup target add thumbv6m-none-eabi --toolchain=nightly
+    - rustc --version
+  provider_api_test_script:
+    - cargo +nightly build --target thumbv6m-none-eabi --no-default-features --features=rust_1_81,unstable-provider-api
+  before_cache_script: rm -rf $CARGO_HOME/registry/index
+
 nightly_test_task:
   name: "Rust Nightly"
   container:

--- a/src/report.rs
+++ b/src/report.rs
@@ -245,7 +245,7 @@ impl<'a> fmt::Display for ReportFormatter<'a> {
             use crate::error;
 
             if let Some(bt) = error::request_ref::<crate::Backtrace>(self.0) {
-                std::writeln!(f, "\nBacktrace:\n{}", bt)?;
+                writeln!(f, "\nBacktrace:\n{}", bt)?;
             }
         }
 

